src/promptflow/promptflow/_sdk/data/tool.schema.json

```diff
@@ -0,0 +1,394 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Tool",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "type": {
+      "$ref": "#/definitions/ToolType"
+    },
+    "inputs": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/InputDefinition"
+      }
+    },
+    "outputs": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/OutputDefinition"
+      }
+    },
+    "description": {
+      "type": "string"
+    },
+    "connection_type": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ConnectionType"
+      }
+    },
+    "module": {
+      "type": "string"
+    },
+    "class_name": {
+      "type": "string"
+    },
+    "source": {
+      "type": "string"
+    },
+    "LkgCode": {
+      "type": "string"
+    },
+    "code": {
+      "type": "string"
+    },
+    "function": {
+      "type": "string"
+    },
+    "action_type": {
+      "type": "string"
+    },
+    "provider_config": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/InputDefinition"
+      }
+    },
+    "function_config": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/InputDefinition"
+      }
+    },
+    "icon": {},
+    "category": {
+      "type": "string"
+    },
+    "tags": {
+      "type": "object",
+      "additionalProperties": {}
+    },
+    "is_builtin": {
+      "type": "boolean"
+    },
+    "package": {
+      "type": "string"
+    },
+    "package_version": {
+      "type": "string"
+    },
+    "default_prompt": {
+      "type": "string"
+    },
+    "enable_kwargs": {
+      "type": "boolean"
+    },
+    "deprecated_tools": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    },
+    "tool_state": {
+      "$ref": "#/definitions/ToolState"
+    }
+  },
+  "definitions": {
+    "ToolType": {
+      "type": "string",
+      "description": "",
+      "x-enumNames": [
+        "Llm",
+        "Python",
+        "Action",
+        "Prompt",
+        "CustomLLM",
+        "CSharp"
+      ],
+      "enum": [
+        "llm",
+        "python",
+        "action",
+        "prompt",
+        "custom_llm",
+        "csharp"
+      ]
+    },
+    "InputDefinition": {
+      "type": "object",
+      "additionalProperties": true,
```

Comment:
Change additionalProperties to true to support properties that are not defined in the input
cc @melionel

Reply:
The expected behavior is not only the InputDefinition allow additionalProperties. The whole json for the Tool should allow additionalProperties.